Templating/Helper/NumberHelper.php

```diff
@@ -17,171 +17,267 @@
  * NumberHelper displays culture information.
  *
  * @author Thomas Rabaix <thomas.rabaix@ekino.com>
+ * @author Stefano Arlandini <sarlandini@alice.it>
  */
 class NumberHelper extends BaseHelper
 {
     /**
-     * @var array
+     * @var array The default attributes to apply to the \NumberFormatter instance
      */
     protected $attributes = array();
 
     /**
-     * @var array
+     * @var array The default text attributes to apply to the \NumberFormatter instance
      */
     protected $textAttributes = array();
 
     /**
-     * Constructor.
+     * @var array The symbols used by \NumberFormatter.
+     */
+    protected $symbols = array();
+
+    /**
+     * Class constructor.
      *
-     * @param string                  $charset        The output charset of the helper
-     * @param LocaleDetectorInterface $localeDetector
-     * @param array                   $attributes     The default attributes to apply to the NumberFormatter instance
-     * @param array                   $textAttributes The default text attributes to apply to the NumberFormatter instance
+     * @param string                  $charset        The output charset of the helper.
+     * @param LocaleDetectorInterface $localeDetector A locale detector instance.
+     * @param array                   $attributes     The default attributes to apply to the \NumberFormatter instance.
+     * @param array                   $textAttributes The default text attributes to apply to the \NumberFormatter instance.
+     * @param array                   $symbols        The default symbols to apply to the \NumberFormatter instance.
      */
-    public function __construct($charset, LocaleDetectorInterface $localeDetector, array $attributes = array(), array $textAttributes = array())
+    public function __construct($charset, LocaleDetectorInterface $localeDetector, array $attributes = array(), array $textAttributes = array(), array $symbols = array())
     {
         parent::__construct($charset, $localeDetector);
 
-        $this->attributes       = $attributes;
-        $this->textAttributes   = $textAttributes;
+        $this->attributes = $attributes;
+        $this->textAttributes = $textAttributes;
+        $this->symbols = $symbols;
     }
 
     /**
-     * format a value with its percent representation (0.1 => 10%).
+     * Formats a number as percent according to the specified locale and
+     * \NumberFormatter attributes.
      *
-     * @param float $number
-     * @param array $attributes
-     * @param array $textAttributes
-     * @param null  $locale
+     * @param string|float|int $number         The number to format.
+     * @param array            $attributes     The attributes used by \NumberFormatter
+     * @param array            $textAttributes The text attributes used by \NumberFormatter.
+     * @param array            $symbols        The symbols used by the formatter.
```

Comment:
@Soullivaneuh i think we don't want to keep old/invalid php doc, what do you say?